src/common/reconciler.go

```diff
@@ -229,58 +202,83 @@ func (r *ServiceReconciler) updateService(service *opslevel.Service, registratio
 		}
-	} else if registration.Tier != "" {
-		log.Warn().Msgf("[%s] Unable to find 'Tier' with alias '%s'", service.Name, registration.Tier)
 	}
-	if v, ok := opslevel.Cache.TryGetLifecycle(registration.Lifecycle); ok {
-		if v == nil {
-			err := fmt.Errorf("the cache unexpectedly returned a lifecycle that is nil - please submit a bug report")
-			log.Warn().Msgf("[%s] unexpected happened: %v", service.Name, err)
-		} else {
-			updateServiceInput.LifecycleAlias = opslevel.RefOf(v.Alias)
-		}
-	} else if registration.Lifecycle != "" {
-		log.Warn().Msgf("[%s] Unable to find 'Lifecycle' with alias '%s'", service.Name, registration.Lifecycle)
+	// TODO: alias equivalency - need to check ALL the aliases
+	if registration.System != "" && (service.Parent == nil || registration.System != service.Parent.Aliases[0]) {
+		updateServiceInput.Parent = opslevel.NewIdentifier(registration.System)
 	}
-	if v, ok := opslevel.Cache.TryGetTeam(registration.Owner); ok {
-		if v == nil {
-			err := fmt.Errorf("the cache unexpectedly returned a team that is nil - please submit a bug report")
-			log.Warn().Msgf("[%s] unexpected happened: %v", service.Name, err)
-		} else {
-			updateServiceInput.OwnerInput = opslevel.NewIdentifier(v.Alias)
-		}
-	} else if registration.Owner != "" {
-		log.Warn().Msgf("[%s] Unable to find 'Team' with alias '%s'", service.Name, registration.Owner)
-	}
-	if r.ServiceNeedsUpdate(updateServiceInput, service) {
-		updatedService, updateServiceErr := r.client.UpdateService(updateServiceInput)
-		if updateServiceErr != nil {
-			log.Error().Msgf("[%s] Failed updating service\n\tREASON: %v", service.Name, updateServiceErr.Error())
-		} else if updatedService == nil {
-			log.Warn().Msgf("[%s] unexpected happened: updated service but the result is nil - please submit a bug report", service.Name)
-		} else if diff := cmp.Diff(service, updatedService); diff != "" {
-			log.Info().Msgf("[%s] Updated Service - Diff:\n%s", service.Name, diff)
+	if registration.Product != "" && registration.Product != service.Product {
+		updateServiceInput.Product = opslevel.RefOf(registration.Product)
+	}
+	if registration.Tier != "" && registration.Tier != service.Tier.Alias {
+		if tier, ok := opslevel.Cache.TryGetTier(registration.Tier); ok {
+			if tier == nil {
+				err := fmt.Errorf("the cache unexpectedly returned a tier that is nil - please submit a bug report")
+				log.Warn().Msgf("[%s] unexpected happened: %v", service.Name, err)
+			} else {
+				updateServiceInput.TierAlias = opslevel.RefOf(tier.Alias)
+			}
+		} else if registration.Tier != "" {
+			log.Warn().Msgf("[%s] Unable to find 'Tier' with alias '%s'", service.Name, registration.Tier)
 		}
-	} else {
+	}
+	// if there is nothing in updateServiceInput aside from the service ID, do not send an update service API call
+	inputDiff := cmp.Diff(opslevel.ServiceUpdateInput{Id: &service.Id}, updateServiceInput)
```

Comment:
this don't feels like the right kind of diffing.  But i see why we cannot compare `Service` to `UpdateServiceInput` - could we make JSON comparisons?

Another option would be to turn a `Service` into a `UpdateServiceInput` and do the compare then.